src/libtomahawk/accounts/DelegateConfigWrapper.cpp

```diff
@@ -227,8 +227,30 @@ DelegateConfigWrapper::onConfigTestResult( Tomahawk::Accounts::ConfigTestResultT
 
         m_invalidData = m_widget->readData();
 
-        // TODO: generate message based on status code
-        m_errorLabel->setText( QString( "<font color='red'>%1</font>" ).arg( tr( "Your config is invalid." ) ) );
+	QString msg;
+	if (message.isEmpty()){
+	    msg = getTestConfigMessage(code);
+	} else {
+	    msg = message;
+	}
+        m_errorLabel->setText( QString( "<font color='red'>%1</font>" ).arg( msg ) );
+    }
+}
+
+QString
+DelegateConfigWrapper::getTestConfigMessage( int code )
+{
+    switch(code) {
+	case Tomahawk::Accounts::ConfigTestResultCommunicationError:
```

Comment:
I'm not sure every compiler likes switching an integer and casing against enum.. possibly an explicit cast would be a good idea here.